src/core/load-fragment.c

```diff
@@ -1277,6 +1277,9 @@ int config_parse_numa_mask(const char *unit,
                 return 0;
         }
 
+        if (isempty(rvalue))
```

Comment:
instead of testing the string here, please change parse_cpu_set_extend() to return either 0 or 1 indicating whether entries where added or the list reset. i.e. return 0 if the list was reset and 1 if the list was extended. Then use that here.

We have functions that return 0 and 1 in different cases of success at many places, this would be natural here too.

It sucks having parsers for the same strings at two places, which you do right now, and with such an approach the parsing is monopolized in parse_cpu_set_extend() which is good.